lib/phoenix_ecto/html.ex

```diff
@@ -231,17 +231,25 @@ if Code.ensure_loaded?(Phoenix.HTML) do
     defp form_for_errors(%{action: nil}), do: []
     defp form_for_errors(%{errors: errors}), do: errors
 
-    defp form_for_hidden(data) do
-      # Since they are primary keys, we should ignore nil values.
-      for {k, v} <- Ecto.primary_key(data), v != nil, do: {k, v}
+    defp form_for_hidden(%{__struct__: module} = data) do
+      if Keyword.has_key?(module.__info__(:functions), :__schema__) do
```

Comment:
If we're going to do this, we should use `function_exported?/3`. I'm not convinced, though.